lib/OpenQA/Schema/ResultSet/Assets.pm

```diff
@@ -28,9 +28,17 @@ sub register ($self, $type, $name, $options = {}) {
         log_warning "no file found for asset '$name' type '$type'";
         return undef;
     }
-    $self->result_source->schema->txn_do(
+    my $schema = $self->result_source->schema;
+    my $sth = $schema->storage->dbh->prepare_cached(
+        <<'END_SQL'
+        INSERT INTO assets (type, name, t_created, t_updated)
+                    VALUES (?,    ?,    now(),     now()    ) ON CONFLICT DO NOTHING
+END_SQL
```

Comment:
If we can depend on Perl 5.16 this could use an indented heredoc and look much nicer.
```perl
    my $sth = $schema->storage->dbh->prepare_cached(<<~'END_SQL');
        INSERT INTO assets (type, name, t_created, t_updated)
                    VALUES (?,    ?,    now(),     now()    ) ON CONFLICT DO NOTHING
    END_SQL
```